scripts/tests/create-flow-from-icmp-error.sh

```
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+source "./scripts/tests/lib.sh"
+
+# Authorize resource 1
+client_curl_resource "172.20.0.100/get"
+client_curl_resource "[172:20:0::100]/get"
+
+# Authorize resource 2 (important, otherwise the Gateway will close the connection on the last resource being removed)
+client_ping_resource example.com
+
+# Revoke access to resource 1
+api_send_reject_access "mycro-aws-gws" "MyCorp Network"        # This is the 172.20.0.1/16 network
+api_send_reject_access "mycro-aws-gws" "MyCorp Network (IPv6)" # This is the 172:20:0::1/64 network
+
+# Try to access resource 1 again
+# First one for each IP will fail because we get an ICMP error.
+client_curl_resource "172.20.0.100/get" || true
+client_curl_resource "[172:20:0::100]/get" || true
```

The nature of ICMP errors is that they will still fail the initial connection but the subsequent ones will go through without issues.

---

this code doesn't seem to check that accessing the resource failed, it only silences the error

I'll throw in a small commit that deals with this, ideally something like this (will expand our integration library) 

`expect_error client_curl_resource "172.20.0.100/get"`

---

@thomaseizinger I've pushed a commit that uses the new functionality, along with some unit test coverage of the helper & adding it to CI. The latter is a bit awkward as I didn't want to create a whole new stage of CI, so I piggybacked it into static analysis part of the workflows. Feel free to move it. :)

---

Thanks! I hope you don't mind that I've removed those tests again. I don't think we need maintain an entire bash script to test a 5 line bash function. It seems out of balance in terms of the amount of code we have in here vs the risk that these tests eliminate.

Bash is notorious for how bad it is and how many pitfalls it has. Testing bash with bash doesn't seem like a good idea because I now need to review even more bash code to make sure it is actually correct.

---

all good;  I'll add a dedicated PR for these, with proper argumentation. I agree that we shouldn't just slip them here. and then we can decide to not merge anyway. 

the main thing is the actual functionality, and I guess it's tested in the integration test.